Stratis.Bitcoin/Features/Wallet/WalletManager.cs

```diff
@@ -1193,13 +1206,45 @@ public TransactionFoundEventArgs(Script script, uint256 transactionHash)
         }
     }
 
-    public class UnspentInfo
+    /// <summary>
+    /// Represents an UTXO that keeps a reference to <see cref="HdAddress"/> and <see cref="HdAccount"/>.
+    /// </summary>
+    /// <remarks>
+    /// This is useful when an UTXO needs access to its HD proprieties like the HD path when recosntructing a provate key.
+    /// </remarks>
+    public class UnspentOutputReference
     {
+        /// <summary>
+        /// The account associated with this UTXO
+        /// </summary>
         public HdAccount Account { get; set; }
-
+ 
+        /// <summary>
+        /// The address associated with this UTXO
+        /// </summary>
         public HdAddress Address { get; set; }
 
-        public List<TransactionData> Transactions { get; set; }
+        /// <summary>
+        /// The transaction representing the UTXO.
+        /// </summary>
+        public TransactionData Transaction { get; set; }
+    }
+
+
+    /// <summary>
+    /// Represent a high level account container that hold all its <see cref="UnspentOutputReference"/>.
+    /// </summary>
+    public class UnspentAccountReference
```

Comment:
It does only contain 1 account, all the child Accounts in UnspentOutputReference are just the same account to provide an easy mapping between Transaction => Address => Account